### PR TITLE
AP-79: Implement documentation generator & project site

### PR DESCRIPTION
**Pull request by chrisdevereux**

### CODE REVIEW

File: site/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "bright-js-framework-docs",
+  "private": true,
+  "description": "Documentation for React Serve",
```
Comment (benbrowning) [inline site/package.json]: @chrisdevereux Should this be 'Bright JS Framework' instead of React-Serve?

File: gulpfile.js [hunk 2]
```diff
@@ -2,13 +2,15 @@ const gulp = require('gulp')
 const del = require('del')
 const shell = require('gulp-shell')
 const ts = require('gulp-typescript')
+const typedoc = require('gulp-typedoc')
 const mocha = require('gulp-mocha')
 const tslint = require('gulp-tslint')
 const distFiles = require('./package.json').files
 
-const PRIVATE_INTERFACE_FILES = ["./src/lib/**/*.d.ts"]
 const SOURCE_FILES = ["./src/**/*.ts", "./src/**/*.tsx"]
```
Comment (richard-hulm) [inline gulpfile.js:10]: src/**/*.ts?(x)

Is (if this is a regex?) equivalent to duplicating between checking ts and not ts

Also, on other projects I've had some files missed using this. I'm unsure if ./src/**/*.ts matches a file in src E.g.

/src/hello.ts - would this be caught by this? I know on other projects I've had to do 'src/*.ts?(x)' 'src/**/*.ts?(x)' to get it to work..

File: site/src/pages/404.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+
+const NotFoundPage = () => (
+  <div>
+    <h1>NOT FOUND</h1>
+    <p>You just hit a route that doesn&#39;t exist... the sadness.</p>
```
Comment (richard-hulm) [inline site/src/pages/404.js]: Woe is the user

**Review by richard-hulm — DISMISSED** [review]

File: site/package.json (hunk 1 above)
Comment (chrisdevereux) [reply site/package.json]: 👍

**Review by richard-hulm — DISMISSED** [review]

**Comment by robotdevil** [comment]
General question, probably my ignorance - is the 'site' folder purely the documentation site? If so, does it have to be named 'site' or could it be made more specific ('documentation' / 'docs' / 'doc-site') so that new devs can easily intuit what it is? Is 'site' convention?

File: site/gatsby-node.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  // Add a page for each documented module, passing through the documentation data generated from
+  // source as a property of the page's pathContext.
+  //
+  // See: https://www.gatsbyjs.org/docs/creating-and-modifying-pages/
+
+  const json = require('./docs.json')
+
+  json.children.forEach((moduleDocJson) => {
+    createPage({
+      path: `/modules/${kebabCase(moduleDocJson.name)}`,
```
Comment (robotdevil) [inline site/gatsby-node.js]: This is the first time I've encountered the term 'kebabCase' - good stuff - probably even better after a few beers...

Comment (chrisdevereux) [reply site/gatsby-node.js]: --yes-it-is--
🔥🔥🔥🔥

Comment (wheelr) [reply site/gatsby-node.js]: +1
The game has changed.

File: site/package.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "keywords": [
+    "gatsby"
+  ],
+  "license": "MIT",
+  "main": "n/a",
+  "scripts": {
+    "build": "gatsby build --prefix-paths",
+    "develop": "gatsby develop",
+    "format": "prettier --trailing-comma es5 --no-semi --single-quote --write 'src/**/*.js'",
```
Comment (wheelr) [inline site/package.json]: I really like this approach to standardising code formatting. Is this a manual invocation?

**Review by wheelr — APPROVED** [review]

**Review by robotdevil — APPROVED** [review]

File: site/package.json (hunk 5 above)
Comment (chrisdevereux) [reply site/package.json]: This is part of the docsite starter kit and only applies to that subdirectory. There's a separate PR open to add a githook that runs a code formatter on the framework codebase, though

File: gulpfile.js (hunk 2 above)
Comment (chrisdevereux) [reply gulpfile.js:10]: With gulp, `**` in a path pattern will match same directory, so including "./src/lib/*.d.ts" isn't necessary. I had to confirm that though, so good question!

**Review by richard-hulm — APPROVED** [review]

**Pull request merged**